compiler/p/codegen/BinaryEvaluator.cpp

```diff
@@ -2058,7 +2058,7 @@ strengthReducingLongDivideOrRemainder32BitMode(TR::Node *node,      TR::CodeGene
       {
       TR::LabelSymbol *callLabel = generateLabelSymbol(cg);
 
-      if (secondChild->getOpCode().isLoadConst() && drConst<=(int64_t)TR::getMaxSigned<TR::Int32>() && drConst>=(int64_t)TR::getMinSigned<TR::Int32>() && drConst!=-1)
+      if (secondChild->getOpCode().isLoadConst() && drConst<=(int64_t)TR::getMaxSigned<TR::Int32>() && drConst>=(int64_t)TR::getMinSigned<TR::Int32>() && drConst!=-1 && drConst!=0)
```

Comment:
there is mandatory divByZero exception.  I don't see the necessity to check !=0 here.

Reply:
You're right that the code generated here cannot be reached if `drConst == 0`. However, we still end up generating code for performing the division which ends up making its way to the binary encoder. That code ends up being garbage and containing instructions with invalid immediates, causing the new binary encoder to trigger an assertion failure since the instructions it's trying to encode are invalid. This change is purely to silence that failure, since the actual code can never be executed in practice.